tests/file_system_cache_test.py

```diff
@@ -56,5 +56,69 @@ def test_empty_key_get(self):
      with self.assertRaisesRegex(ValueError , r"key cannot be empty"):
        cache.get("")
 
+  def test_size_of_directory(self):
+    with tempfile.TemporaryDirectory() as tmpdir:
+      cache = FileSystemCache(tmpdir)
+      cache.put("foo", b"bar")
+      self.assertEqual(cache._get_cache_directory_size(), 3)
+
+  def test_size_of_empty_directory(self):
+    with tempfile.TemporaryDirectory() as tmpdir:
+      cache = FileSystemCache(tmpdir)
+      self.assertEqual(cache._get_cache_directory_size(), 0)
+
+  def test_size_of_existing_directory(self):
+    with tempfile.TemporaryDirectory() as tmpdir:
+      cache1 = FileSystemCache(tmpdir)
+      cache1.put("foo", b"bar")
+      del cache1
+      cache2 = FileSystemCache(tmpdir)
+      self.assertEqual(cache2._get_cache_directory_size(), 3)
+
+  def test_cache_is_full(self):
+    with tempfile.TemporaryDirectory() as tmpdir:
+      cache = FileSystemCache(tmpdir, max_cache_size_bytes=5)
+      cache.put("foo", b"bar")
+      cache.put("second", b"two")
+      self.assertEqual(cache.get("foo"), None)
+
+  def test_file_removed(self):
+    with tempfile.TemporaryDirectory() as tmpdir:
+      cache = FileSystemCache(tmpdir, max_cache_size_bytes=5)
+      cache.put("foo", b"bar")
+      cache.put("second", b"two")
+      self.assertEqual(cache.get("second"), b"two")
+
+  def test_delete_multiple_files(self):
+    with tempfile.TemporaryDirectory() as tmpdir:
+      cache = FileSystemCache(tmpdir, max_cache_size_bytes=6)
+      cache.put("foo", b"bar")
+      cache.put("second", b"two")
+      cache.put("third", b"three")
+      self.assertEqual(cache.get("second"), None)
+
+  def test_big_file_exists(self):
+    with tempfile.TemporaryDirectory() as tmpdir:
+      cache = FileSystemCache(tmpdir, max_cache_size_bytes=6)
+      cache.put("foo", b"bar")
+      cache.put("second", b"two")
+      cache.put("third", b"three")
+      self.assertEqual(cache.get("third"), b"three")
+
```

Comment:
Also add tests to make sure that when there are multiple files that could be removed to make room, the least-recently-accessed file is removed. Create a test case for each of these:

1. Adding at least 3 files until one needs to be removed. Make sure that the first one added is removed, and that the other ones are still there.
2. Adding at least 2 files, `get` the first one added, and then add more file(s) until one needs to be removed. Assert that the least-recently-accessed one is deleted (i.e. not the one you called `get` on), and that the others are still there.